lib/Dialect/SV/Transforms/HWMemSimImpl.cpp

```diff
@@ -190,6 +253,73 @@ void HWMemSimImplPass::runOnOperation() {
     if (genOp.descriptor() == "FIRRTL_Memory") {
       auto mem = analyzeMemOp(oldModule);
 
+      // If this memory has defined write-after-write behavior, we need to
```

Comment:
Does SFC do a whole program analysis like this or just a local comparison?

Reply:
No. It does not. This was necessary here because generator ops could deduplicate. However, with the new approach of putting the clock information into the generator op, this is unnecessary.

To determine clock equivalence, the SFC is building a module-local netlist during Verilog emission that looks through connects and nodes.  I'm looking into replicating that behavior here.

Reply:
I double checked this. Clock equivalence is a trivial "what is the first connection to the clock". This means that something trivial can change the write order behavior, e.g., you can cause the SFC to emit a different memory with a `node tmp = clock` that has a dont touch annotation.

The behavior that's being used in this PR just happens to be the same behavior. I'll lock this in with updates to the `lower-to-hw.mlir`.